src/Controller/ComposerController.php

```diff
@@ -66,9 +66,10 @@ public function inspect(ServerRequestInterface $request, Aliases $aliases): Resp
 
     public function require(ServerRequestInterface $request, Aliases $aliases): ResponseInterface
     {
-        $package = $request->getParsedBody()['package'] ?? null;
-        $version = $request->getParsedBody()['version'] ?? null;
-        $isDev = $request->getParsedBody()['isDev'] ?? false;
+        $parsedBody = \json_decode($request->getBody()->getContents(), true, 512, JSON_THROW_ON_ERROR);
```

Comment:
Why didn't `getParsedBody()` work for you?

Reply:
I don't know, seems something broken in `yii-runner-http`

Reply:
[ServerRequestFactory](https://github.com/yiisoft/yii-runner-http/blob/master/src/ServerRequestFactory.php) from runner used array `$_POST` for fill parsed body, but in apps and [demo](https://github.com/yiisoft/demo/blob/master/blog-api/config/common/di/psr17.php#L7) used [ServerRequestFactory](https://github.com/httpsoft/http-message/blob/master/src/ServerRequestFactory.php) that not parse body and use `null` for it.

We can't rely on parsed body here.

Reply:
See https://github.com/yiisoft/yii-debug-api/pull/118#pullrequestreview-1505427738